docs/rfcs/0004-cirrus-experiments.md

```diff
@@ -0,0 +1,260 @@
+
+Once the feature list is received from Cirrus, it will be parsed and known features will be turned on while unknown features will be discarded.
+
+Since the Cirrus features will be served alongside the list of features on the featurelist endpoint, TaskGetFeatureList will also be reused to apply experiment features. 
+
+Experimental features may have variables associated with them as can be seen by the [Cirrus feature API response](https://github.com/mozilla/experimenter/tree/main/cirrus#output). Each experimental feature will have a settings group associated with it, with the exact same name as the feature. Each variable will have a corresponding setting under the group associated with it, with the exact same name. For example, an experimental feature like so:
+
+```
+{
+
+ "featuresOverwrite":  {  },
+ "experimentalFeatures":  {
+    "onboardingCtaStyleExperiment":  {
+      "color":  "#FF0000",
+      "fontFamily":  "Proxima Nova"
+    },
+  }
+}
+```
+
+Will have a related feature like so:
+
+```
+EXPERIMENTAL_FEATURE(
+      onboardingCtaStyleExperiment, // Feature ID
+      "Experimentations on the onboarding CTA", // Feature name
+      QStringList("color", "fontFamily"),  // Feature variables
+)
+```
+
+The EXPERIMENTAL_FEATURE macro, intentionally doesn't contain the field for declaring if the feature can be flipped on or off. Experimental features must always be "toggleable".
+
+Note Mozilla VPN features may or may not be "toggleable", meaning they may not be allowed to be turned on or off. Experimental features must always be toggleable, otherwise they cannot be applied dynamically. This has consequences for the implementation of the treatment itself. This needs to be considered on a case by case basis.
+
+It also doesn't contain the feature callback field. Experimental features always default to false unless they have been turned on through Cirrus. 
+
+Finally, the experimental feature settings will be accessible from C++ like so:
+
+```
+settingsHolder.onboardingCtaStyleExperiment.color()
+settingsHolder.onboardingCtaStyleExperiment.fontFamily()
+```
+
+Or from QML like so:
+
+```
+MZSettings.onboardingCtaStyleExperiment.color
+MZSettings.onboardingCtaStyleExperiment.fontFamily
+```
+
+Experimental feature settings will only have public getters. They may only be set or removed by the TaskGetFeatureList.
+
+Note In order for this setup to work, experimental features will have to be declared in a separate feature list file e.g. experimentalfeaturelist.h. This will make it possible to import the experimental features only on the settings holder module and create the settings API necessary.
```

Comment:
```suggestion
> **Note** In order for this setup to work, experimental features will have to be declared in a separate feature list file e.g. experimentalfeaturelist.h. This will make it possible to import the experimental features only on the settings holder module and create the settings API necessary.
```